### ISSUE TITLE
[Accessibility: ACC-264.1] Button: Incorrect UI Type read

### ISSUE THREAD

**Issue by divyamalhotra488** [SAP/fundamental-ngx#12614]
#### Is this a bug, enhancement, or feature request?
Bug

#### Describe your proposal.
Type of the UI should be read out correctly to screen reader users.
Providing role, state, and value information on all user interface components enables compatibility with assistive technology, such as screen readers, screen magnifiers, and speech recognition software, used by people with disabilities.

#### Can you handle that on the application side
No

#### Which versions of Angular and Fundamental Library for Angular are affected? Please, specify the exact version. (If this is a feature request, use current version.)
All

#### If this is a bug, please provide steps for reproducing it; the exact components you are using;
Component used: [Dynamic Page](https://sap.github.io/fundamental-ngx/#/core/dynamic-page)
UI type for ‘Collapse Header’ button is read as ‘Button menu’ which is incorrect.
<img width="452" alt="image" src="https://github.com/user-attachments/assets/1bf876c3-b7a2-4f72-82e5-d26e3deeaa0a">

#### Please provide stackblitz example(s).
You can try ngx examples, issue is reproducible there.

#### In case this is Accessibility related topic, did you consult with an accessibility expert? If not, please do so and share their recommendations.
Yes, issue is raised by an accessibility expert.

#### Did you check the documentation and the API?
Yes

#### Did you search for similar issues?
Yes

#### Is there anything else we should know?

**_IMPORTANT: Please refrain from providing links or screenshots of SAP's internal information, as this project is open-source, and its contents are accessible to anyone._**
